File: repository/src/main/java/com/sreMake/repository/system/ResourcesRepository.java
package com.sreMake.repository.system;

import com.sreMake.model.system.Resources;
import org.babyfish.jimmer.spring.repo.JavaRepository;

import java.util.List;

public interface ResourcesRepository extends JavaRepository<Resources, Long> {

    /**
     * 通过路由路径获取Resources
     */
    List<Resources> listByRouters(String... routers);

    /**
     * parent 获取全部相关的父级
     */
    List<Resources> listParentId(Long... parentIds);
}

File: repository/src/main/kotlin/com/SReMake/repository/system/ResourcesRepository.kt
package com.SReMake.repository.system

import com.SReMake.model.system.Resources
import org.babyfish.jimmer.spring.repo.KotlinRepository

interface ResourcesRepository : KotlinRepository<Resources, Long> {
    /**
     * 通过路由路径获取Resources
     */
    fun listByRouters(routers: Collection<String>): List<Resources>

    /**
     * parent 获取全部相关的父级
     */
    fun listParentId(parentIds: Collection<Long>): List<Resources>
}